osdep/timer.c

```diff
@@ -73,44 +72,41 @@ int64_t mp_time_ns_add(int64_t time_ns, double timeout_sec)
     return time_ns + ti;
 }
 
-#if !HAVE_WIN32_INTERNAL_PTHREADS
-static int get_realtime(struct timespec *out_ts)
+static void get_realtime(struct timespec *out_ts)
 {
 #if defined(_POSIX_TIMERS) && _POSIX_TIMERS > 0
-    return clock_gettime(CLOCK_REALTIME, out_ts);
+    clock_gettime(CLOCK_REALTIME, out_ts);
 #else
     // OSX
     struct timeval tv;
     gettimeofday(&tv, NULL);
     out_ts->tv_sec = tv.tv_sec;
     out_ts->tv_nsec = tv.tv_usec * 1000UL;
-    return 0;
 #endif
 }
-#endif
 
 struct timespec mp_time_ns_to_realtime(int64_t time_ns)
 {
-    struct timespec ts = {0};
-
-#if !HAVE_WIN32_INTERNAL_PTHREADS
-    if (get_realtime(&ts) != 0)
-        return ts;
-    int64_t time_rel = time_ns - mp_time_ns();
-#else
-    int64_t time_rel = time_ns;
-#endif
-
-    // clamp to 1000 days in the future
-    time_rel = MPMIN(time_rel, 1000 * 24 * 60 * 60 * INT64_C(1000000000));
-    ts.tv_sec += time_rel / INT64_C(1000000000);
-    ts.tv_nsec += time_rel % INT64_C(1000000000);
-
-    if (ts.tv_nsec >= INT64_C(1000000000)) {
-        ts.tv_sec++;
-        ts.tv_nsec -= INT64_C(1000000000);
+    struct timespec ts;
+    get_realtime(&ts);
+    // We don't know what time source mp_time_ns() uses, but usually it's not
+    // CLOCK_REALTIME - so we have to remap the times.
+    int64_t unow = mp_time_ns() / 1000L;
```

Comment:
Truncates the time to microseconds. I think we don't want that given all our code were converted to nanoseconds recently?